src/shared/containers/challenge-detail/index.jsx

```diff
@@ -148,11 +148,16 @@ class ChallengeDetailPageContainer extends React.Component {
   componentWillReceiveProps(nextProps) {
     const userId = _.get(this, 'props.authTokens.user.userId');
     const nextUserId = _.get(nextProps, 'authTokens.user.userId');
+    const {
+      tokenV3,
+      authTokens,
+    } = this.props;
     if (userId !== nextUserId) {
       nextProps.getCommunitiesList(nextProps.authTokens);
     }
 
-    if (this.props.tokenV3 !== nextProps.tokenV3) {
+    if (tokenV3 !== nextProps.tokenV3
+      || authTokens.user.userId !== nextProps.authTokens.user.userId) {
```

Comment:
It is not guaranteed that `authTokens.user` is always defined, that's why above in this function `_.get(..)` is used to fetch `userId` and `nextUserId` from `this.props` and `nextProps`. Anyway, if you compare auth tokens, then there is no need to append `|| TOKENS_ARE_DIFFERENT`: user ID is in fact encoded into the token, thus if the user IDs are different, then tokens are different as well.

There is one problem with your fix:
- When the app is opened, the v3 auth token is automatically refreshed about each 10 mins;
- Hence, your change will cause reloading of challenge details each 10 mins;
- Actually, we used to reload it this way before and it had a bad side-effect: if an user had opened challenge details and work on them, and then something happend with his Internet connection, or there is some instability of the TC backend, within 10 mins the challenge details reload will fail, and the already loaded challenge details will be dropped from the app.

So, to avoid that problem we should either (1) ensure that the details are reloaded only once, when we really need it to check the downloads; (2) to ensure that if reloading fails, but the challenge details have been already loaded before, then we don't drop them, and the code does not crush. We also should inform the user about problems with the reloading with the help of the API or network error icons (see the issue https://github.com/topcoder-platform/community-app/issues/611 and the related PR/commits already accepted into the codebase)